File: 201. Bitwise AND of Numbers Range/Python/Solution.py
class Solution(object):
    def rangeBitwiseAnd(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        the_xor = m ^ n
        diff_index = 31
        while diff_index >= 0 and not the_xor & 1 << diff_index:
            diff_index -= 1
        return m & 0xffffffff << (diff_index + 1)

File: 201. Bitwise AND of Numbers Range/C++/Solution.cpp
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        int theXor = m ^ n;
        int diffIndex = 31;
        while (diffIndex >= 0 && (theXor & 1 << diffIndex) == 0) {
            diffIndex--;
        }
        return m & 0xffffffff << (diffIndex + 1);
    }
};